docs/current/cookbook/snippets/build-dockerfile/index.mjs

```diff
@@ -0,0 +1,24 @@
+import { connect } from "@dagger.io/dagger"
+
+connect(
+  async (client) => {
+    // get build context directory
+    const contextDir = client.host().directory("/projects/myapp")
+
+    // get Dockerfile in different filesystem location
+    const dockerfilePath = "/data/myapp/custom.Dockerfile"
+    const dockerfile = client.host().file(dockerfilePath)
+
+    // add Dockerfile to build context directory
+    const workspace = contextDir.withFile("custom.Dockerfile", dockerfile)
+
+    // build using Dockerfile
+    // publish the resulting container to a registry
+    const imageRef = await client
+      .container()
+      .build(workspace, {dockerfile: "custom.Dockerfile"})
```

Comment:
I think workspace here needs to be `contexDir` instead of `workspace` since it expects a `Directory` type and not a `File` one.